src/main/java/de/zalando/aruha/nakadi/config/JsonConfig.java

```diff
@@ -33,14 +39,15 @@
     @Bean
     @Primary
     public ObjectMapper jacksonObjectMapper() {
-        final ObjectMapper objectMapper = new ObjectMapper()
-                .setPropertyNamingStrategy(CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES);
+        final ObjectMapper objectMapper = new ObjectMapper();
 
+        objectMapper.setPropertyNamingStrategy(CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES);
         objectMapper.registerModule(enumModule());
+        objectMapper.registerModule(eventTypeSchemaModule());
```

Comment:
In my view it's either or. And I would support having it as a json object (there's no point in keeping a string except for backwards compatibility).
In any of the cases, clarification on API is needed.